face/infra/roc-v3/src/main/java/com/simprints/face/infra/rocv3/detection/RocV3Detector.kt

```diff
@@ -0,0 +1,165 @@
+package com.simprints.face.infra.rocv3.detection
+
+import ai.roc.rocsdk.embedded.SWIGTYPE_p_float
+import ai.roc.rocsdk.embedded.SWIGTYPE_p_unsigned_char
+import ai.roc.rocsdk.embedded.roc_detection
+import ai.roc.rocsdk.embedded.roc_image
+import ai.roc.rocsdk.embedded.roc_landmark
+import android.graphics.Bitmap
+import android.graphics.Rect
+import com.simprints.core.ExcludedFromGeneratedTestCoverageReports
+import com.simprints.face.infra.basebiosdk.detection.Face
+import com.simprints.face.infra.basebiosdk.detection.FaceDetector
+import java.nio.ByteBuffer
+import javax.inject.Inject
+import javax.inject.Singleton
+import ai.roc.rocsdk.embedded.roc as roc3
+
+
+@ExcludedFromGeneratedTestCoverageReports(
+    reason = "This class uses roc 3 class that has native functions and can't be mocked"
+)
+@Singleton
+class RocV3Detector @Inject constructor() : FaceDetector {
+
+    override fun analyze(bitmap: Bitmap): Face? {
+        val rocColorImage = roc_image()
+        val rocGrayImage = roc_image()
+        val byteBuffer = bitmap.toByteBuffer()
+        roc3.roc_from_rgba(
+            byteBuffer.array(),
+            bitmap.width.toLong(),
+            bitmap.height.toLong(),
+            bitmap.rowBytes.toLong(),
+            rocColorImage
+        )
+        roc3.roc_bgr2gray(rocColorImage, rocGrayImage)
+        return detectFace(rocColorImage, rocGrayImage, bitmap.width, bitmap.height)
+    }
+
+    /*
+    To detect a face from image
+    1- use the gray image to detect the face
+    2- if the face is detected, then use the gray image to generate the landmarks
+    3- use the landmarks and the color image to generate the face template
```

Comment:
Interesting! Is this the recommended way to do things by ROC?

Reply:
That is how they do it in the samples. But not written in the docs anywhere